### DISCUSSION TITLE
When executing a UDP to the mle-eid IP, a "No Route" error occurred.

### DISCUSSION THREAD

**Discussion by jiemincao** [openthread/openthread#11052] [Q&A]
When using the UDP, the leader discovered that certain routes were unreachable. After checking the logs, a "No Route" error was found. Upon further investigation, it was determined that when a child switches to a new parent, its router information is not updated in the leader's EID Cache Table, causing this issue.

The currently known solution is to have the child send a UDP to the leader, which triggers the leader to update its EID Cache Table. However, if no IP message is received from the child, the leader will continue to drop packets. It is unclear whether there are other possible solutions.

We are currently using the Thread Reference version 2023-07-06. Has this issue been resolved in a newer version?

The log is as follows:
[N] MeshForwarder-: Dropping IPv6 UDP msg, len:57, chksum:7258, ecn:no, sec:yes, error:NoRoute, prio:normal
[N] MeshForwarder-:     src:[fd00:db8:0:0:0:ff:fe00:fc00]:5683
[N] MeshForwarder-:     dst:[fd00:db8:0:0:0:0:0:104]:5683

**Comment by aaa147147** [comment]
I have encountered this problem before, and updating to the latest version looks much more stable. I suggest you update to the latest version. 
And it is also best to update the Openthread version of Thread devices to the latest version

**Comment by jiemincao** [comment]
Thank you for the response. I will proceed with the update and testing.

**Comment by jiemincao** [comment]
Hi:
After the update, the previous "No Router" issue is no longer occurring, but a new issue has arisen. After the network is set up, some routers are not directly connected to the leader, even though their signal is strong. In the leader's router table, these routers are shown as "not directly connected" to the leader, but on these routers' end, they show that they can directly connect to the leader. This causes the data they send to be lost due to the leader's MAC address.

**leader:**
| ID | RLOC16 | Next Hop | Path Cost | LQ In | LQ Out | Age | Extended MAC     | Link |
+----+--------+----------+-----------+-------+--------+-----+------------------+------+
|  0 | 0x0000 |       29 |         1 |     3 |      3 |  18 | 00000000000000dc |    1 |
|  4 | 0x1000 |       13 |         1 |     3 |      3 |  53 | 00000000000000ea |    1 |
|  5 | 0x1400 |       40 |         1 |     3 |      3 |  15 | 000000000000008f |    1 |
|  8 | 0x2000 |       55 |         1 |     3 |      3 |  27 | 00000000a0780036 |    1 |
| 12 | 0x3000 |       27 |         1 |     3 |      3 |  20 | 000000000000010b |    1 |
| 13 | 0x3400 |       27 |         1 |     3 |      3 |  23 | 00000000000000cb |    1 |
| 14 | 0x3800 |       13 |         1 |     3 |      3 |  23 | 00000000000000b8 |    1 |
| 15 | 0x3c00 |       47 |         1 |     3 |      3 |  13 | 000000000000009f |    1 |
| 16 | 0x4000 |       12 |         1 |     3 |      3 |  14 | 000000000000008b |    1 |
| 18 | 0x4800 |       63 |         0 |     0 |      0 |   0 | 503832333245a638 |    0 |
| 24 | 0x6000 |       35 |         1 |     3 |      3 |  26 | 00000000a0780094 |    1 |
| 27 | 0x6c00 |       47 |         1 |     3 |      3 |  16 | 0000000000000101 |    1 |
| 28 | 0x7000 |       55 |         1 |     0 |      0 | 203 | 00000000a0780132 |    0 |
| 29 | 0x7400 |       15 |         1 |     3 |      3 |  25 | 00000000000000b5 |    1 |
| 31 | 0x7c00 |       43 |         1 |     3 |      3 |  44 | 00000000a0780089 |    1 |
| 33 | 0x8400 |       27 |         1 |     3 |      3 |  52 | 00000000a0780149 |    1 |
| 35 | 0x8c00 |       14 |         1 |     3 |      3 |  26 | 00000000a0780080 |    1 |
| 37 | 0x9400 |       14 |         1 |     3 |      3 |  26 | 00000000a0780086 |    1 |
| 40 | 0xa000 |        0 |         1 |     3 |      3 |  25 | 0000000000000085 |    1 |
| 41 | 0xa400 |       14 |         1 |     3 |      3 |  56 | 00000000000000c2 |    1 |
| 43 | 0xac00 |       35 |         1 |     3 |      3 |  25 | 00000000000000b9 |    1 |
| 44 | 0xb000 |       35 |         1 |     3 |      3 |  27 | 00000000000000cf |    1 |
| 45 | 0xb400 |       15 |         1 |     3 |      3 |  28 | 00000000000000cd |    1 |
| 47 | 0xbc00 |       13 |         1 |     3 |      3 |  12 | 00000000000000b6 |    1 |
| 48 | 0xc000 |       15 |         1 |     3 |      3 |  18 | 00000000a0780084 |    1 |
| 50 | 0xc800 |       35 |         1 |     3 |      3 |  14 | 00000000a0780096 |    1 |
| 52 | 0xd000 |       15 |         1 |     3 |      3 | 200 | 00000000a0780098 |    1 |
| 54 | 0xd800 |       14 |         1 |     3 |      3 |  23 | 0000000000000093 |    0 |
| 55 | 0xdc00 |       27 |         1 |     3 |      3 |  20 | 00000000000000f9 |    1 |
| 57 | 0xe400 |       14 |         1 |     3 |      3 |  20 | 00000000000000bb |    1 |
| 59 | 0xec00 |       35 |         1 |     3 |      3 |  13 | 00000000000000d1 |    0 |

**leader received ec00 adv RSSI:**
[I] MeshForwarder-: Received IPv6 UDP msg, len:120, chksum:e7a0, ecn:no, from:00000000000000d1, sec:no, prio:net, rss:-21.0
[I] MeshForwarder-:     src:[fe80:0:0:0:200:0:0:d1]:19788
[I] MeshForwarder-:     dst:[ff02:0:0:0:0:0:0:1]:19788
[I] Mle-----------: Receive Advertisement (fe80:0:0:0:200:0:0:d1,0xec00)

**ec00 router:**
| ID | RLOC16 | Next Hop | Path Cost | LQ In | LQ Out | Age | Extended MAC     | Link |
+----+--------+----------+-----------+-------+--------+-----+------------------+------+
|  0 | 0x0000 |       29 |         1 |     3 |      3 |   9 | 00000000000000dc |    1 |
|  4 | 0x1000 |       35 |         1 |     3 |      3 |   1 | 00000000000000ea |    1 |
|  5 | 0x1400 |       18 |         1 |     3 |      3 |   5 | 000000000000008f |    1 |
|  8 | 0x2000 |       55 |         1 |     3 |      3 |  13 | 00000000a0780036 |    1 |
| 12 | 0x3000 |       27 |         1 |     3 |      3 |   3 | 000000000000010b |    1 |
| 13 | 0x3400 |       27 |         1 |     3 |      3 |  44 | 00000000000000cb |    1 |
| 14 | 0x3800 |       13 |         1 |     3 |      3 |   5 | 00000000000000b8 |    1 |
| 15 | 0x3c00 |       18 |         1 |     3 |      3 |   6 | 000000000000009f |    0 |
| 16 | 0x4000 |       18 |         1 |     3 |      3 |  12 | 000000000000008b |    1 |
| 18 | 0x4800 |       27 |         1 |     3 |      3 |   2 | 503832333245a638 |    1 |
| 24 | 0x6000 |       54 |         1 |     0 |      0 |  89 | 00000000a0780094 |    0 |
| 27 | 0x6c00 |       57 |         1 |     3 |      3 |  45 | 0000000000000101 |    1 |
| 28 | 0x7000 |       55 |         1 |     3 |      3 |  45 | 00000000a0780132 |    1 |
| 29 | 0x7400 |       14 |         1 |     3 |      3 |   7 | 00000000000000b5 |    1 |
| 31 | 0x7c00 |       54 |         1 |     3 |      3 |  36 | 00000000a0780089 |    0 |
| 33 | 0x8400 |       48 |         1 |     3 |      3 |   8 | 00000000a0780149 |    1 |
| 35 | 0x8c00 |       13 |         1 |     3 |      3 |  30 | 00000000a0780080 |    1 |
| 37 | 0x9400 |       57 |         1 |     3 |      3 |  12 | 00000000a0780086 |    1 |
| 40 | 0xa000 |       18 |         1 |     3 |      3 |  34 | 0000000000000085 |    1 |
| 41 | 0xa400 |       14 |         1 |     3 |      3 |   6 | 00000000000000c2 |    1 |
| 43 | 0xac00 |       13 |         1 |     3 |      3 |   9 | 00000000000000b9 |    0 |
| 44 | 0xb000 |       35 |         1 |     3 |      3 |   7 | 00000000000000cf |    0 |
| 45 | 0xb400 |       18 |         1 |     3 |      3 |   4 | 00000000000000cd |    1 |
| 47 | 0xbc00 |       13 |         1 |     3 |      3 |  25 | 00000000000000b6 |    0 |
| 48 | 0xc000 |        4 |         1 |     3 |      3 |  12 | 00000000a0780084 |    1 |
| 50 | 0xc800 |       35 |         1 |     3 |      3 |   9 | 00000000a0780096 |    1 |
| 52 | 0xd000 |       18 |         1 |     0 |      0 |  42 | 00000000a0780098 |    0 |
| 54 | 0xd800 |       14 |         1 |     3 |      3 |   3 | 0000000000000093 |    1 |
| 55 | 0xdc00 |       35 |         1 |     3 |      3 |   5 | 00000000000000f9 |    1 |
| 57 | 0xe400 |       13 |         1 |     3 |      3 |  30 | 00000000000000bb |    1 |
| 59 | 0xec00 |       63 |         0 |     0 |      0 |   0 | 00000000000000d1 |    0 |


**Reason for lost:**
[I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:124, seqnum:172, type:Data, src:0xec00, dst:0x4800, sec:yes, ackreq:yes
[I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:50, seqnum:173, type:Data, src:0xec00, dst:0x4800, sec:yes, ackreq:yes

**Reply by jwhui** [reply]
It's not clear why the neighboring routers are not connecting directly. Inspecting the logs can provide more visibility on the issue.

Note that too strong a signal can be problematic for some radios due to receiver saturation issues.

**Reply by jiemincao** [reply]
Hi Jwhui :
Thank you for your reply.
I'm not sure what happens during the networking process, but whenever this issue occurs, I have to restart the router to make it resend the link request to solve the problem.
Is there any other solution?

**Reply by jwhui** [reply]
Without additional visibility, it is difficult to diagnose the issue.

What git commit are you using? Can you provide logs?

**Reply by jiemincao** [reply]
I'm using commit version 971f05f7600c508802a4b4e0b2eadef801272969 on the main branch.
I'm trying to find the logs from when it happened, particularly to understand why the router was removed from the neighbor table. Considering that the network environment consists of 300 FTD nodes in the same office, it's not easy to keep track of everything.

**Reply by jiemincao** [reply]
The reason the Link Request cannot be sent is that the Router State is stuck in the Link Request state.

Upon investigation, after receiving the Advertisement packet, there should have been an opportunity to re-establish the Link, but because the Router State remains in the Link Request state, the connection cannot be restored. A deeper look into the code revealed that when the ScheduleLinkRequest fails to be added, the system does not transition the Router State back to Invalid. Additionally, when the Send Link Request fails due to the inability to acquire a buffer, the Router State is not reset to Invalid, preventing the re-establishment of the Link.

Please assist with confirming this. Thank you!

**Reply by abtink** [reply]
Thanks @jiemincao for raising this and debugging the issue. I just submitted the following PR related to this:
- https://github.com/openthread/openthread/pull/11442
